synapse/storage/schema/main/delta/65/06_msc2409_add_device_id_appservice_stream_type.sql

```diff
@@ -0,0 +1,23 @@
+/* Copyright 2021 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Add a column to track what to_device stream id that this application
+-- service has been caught up to.
+
+-- We explicitly don't set this field as "NOT NULL", as having NULL as a possible
+-- state is useful for determining if we've ever sent traffic for a stream type
+-- to an appservice. See https://github.com/matrix-org/synapse/issues/10836 for
+-- one way this can be used.
```

Comment:
thanks for adding this, but I think it is a bit verbose!

```suggestion
-- NULL indicates that this appservice has never received any to_device messages. This
-- can be used to avoid sending a huge dump of messages at startup.
```

Reply:
This feels like it was more of a note-to-self for future work than a comment really meant for anyone else. I'll keep it in mind for myself :)